magento/magento2
# 33310 fix unsubscribe url in newsletter email template

### Description (*)
I found out that the Resolver can't resolve the subscriber method `getUnsubscriptionLink()` and get a url. I changed it by adding property to subscriber before call transport to send email.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#33310

### Manual testing scenarios (*)
1. Magento Admin -> Customers -> All Customers -> Edit the Customer -> News Letter -> Subscribe to NewsLetter and save
2. Marketing -> News Letter Template -> Add new template
3. Marketing -> News Letter Template -> Actions -> Queue News Letter - edit and Save News Letter
4. News Letter should be triggered to the Customer email ID. If not triggered, run magento cron job: sudo php bin/magento cron:run, emails would be triggered automatically after running cron.

### Contribution checklist (*)
